python/_restclient/hydrotools/_restclient/async_client.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+from aiohttp_client_cache import CachedSession
+import asyncio
+import forge
+from functools import partial
+from inspect import Parameter
+from random import random
+import warnings
+
+__all__ = ["ClientSession"]
+
+RETRY_STATUS_CODES = frozenset({503, 429, 301})
+
+
+def forge_client_session(init):
+    """ Forge CachedSession.__init__ method signature """
+    forged_signature = forge.compose(
+        forge.copy(CachedSession.__init__),
+        forge.insert(
+            [
+                forge.kwarg("retry", default=True, type=bool),
+                forge.kwarg("n_retries", default=3, type=int),
+            ],
+            before=lambda x: x.kind == Parameter.KEYWORD_ONLY,
+        ),
+    )(init)
+    return forged_signature
+
+
+with warnings.catch_warnings():
+    # Ignore aiohttp.ClientSession Inheritance warning
+    warnings.filterwarnings(
+        "ignore", category=DeprecationWarning, message="Inheritance class"
```

Comment:
Can you elaborate on why this is necessary in the comments? I found this issue from 2019 concerning this:

https://github.com/aio-libs/aiohttp/issues/3185

Reply:
Why the comment is necessary? I just wanted to add context to why I ignored the warning. This is the same approach that `aiohttp_client_cache` takes to implement their `CachedSession` object. Personally I think throwing the deprecated warning is a bit overzealous. Of course, software will change and that will affect downstream users, but that's just part of the gig. 

My reasoning for filtering the warning is that the way ive implemented the class, it will always throw a warning to the user however it will not break usability.

Reply:
I meant add why you are ignoring the warning as a comment.

Note: `aiohttp_client_cache` also minimally elaborate on why they ignore the warning and the implications.
https://github.com/JWCook/aiohttp-client-cache/blob/main/aiohttp_client_cache/session.py#L60

Example:
```python
# Ignore aiohttp.ClientSession Inheritance warning
#  This is the same approach that aiohttp_client_cache takes to implement their CachedSession object
```

Reply:
Ah, sorry I understand. Yeah Ill add some context in a comment.